lib/bootloader_setup.pm

```diff
@@ -630,10 +630,16 @@ sub autoyast_boot_params {
         $autoyast_args .= "$proto://10.0.2.1/";
         $autoyast_args .= 'data/' if $ay_var !~ /^aytests\//;
         $autoyast_args .= $ay_var;
+    } elsif ($ay_var =~ /^ASSET_\d+$/) {
+        # In case profile is uploaded as an ASSET we need just filename
+        $ay_var = basename(get_required_var($ay_var));
+        $autoyast_args .= autoinst_url("/assets/other/$ay_var");
```

Comment:
i think you should better use the data_url here. which it already has some handling for the paths. Take a look https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L2070

Reply:
Old code used it and it's broken as of now. If data_url would work we would not need hotpatch. I will propose change to data_url too, so we can revert this later, but to unblock testing on TW we need this change.

Reply:
Actually, it seems that the logic should moved to there https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L2070

Reply:
It was not moved, it was not adjusted to the latest changes. `ASSET_*` contains full path, here https://openqa.opensuse.org/tests/1217573#step/installation/43 you can see which url we get. That's why I'm saying that we should rather fix it in os-autoinst. @mloviska has provided proposal, but closed PR which I think we should have.

Let me know if you have more questions.

Reply:
> It was not moved, it was not adjusted to the latest changes. `ASSET_*` contains full path, here https://openqa.opensuse.org/tests/1217573#step/installation/43 you can see which url we get. That's why I'm saying that we should rather fix it in os-autoinst. @mloviska has provided proposal, but closed PR which I think we should have.
> 
> Let me know if you have more questions.

sry, I have thought it is not needed anymore. https://github.com/os-autoinst/os-autoinst/pull/1381

Reply:
@mloviska it depends on point of view, as it's faster to hot patch the tests, but I believe current implementation of `data_url` is broken for ASSETS handling, so I believe it's worth applying your proposal. Thanks!